### ISSUE TITLE
polkadot-js extension is not injected on page reload

### ISSUE THREAD

**Issue by nilathj** [polkadot-js/extension#938]
I'm using the example from the cookbook to get a list of addresses using a simple react app.  It works fine on the 1st load.  When I reload the current page, most of the time, I don't see the list of addresses, it just prints my "no extension found" message.  How can I get the polkadot-js extension to be injected on page reloads?

```
import React, {useState, useEffect} from "react";
import {isWeb3Injected, web3Accounts, web3Enable,} from "@polkadot/extension-dapp";
import type { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';

import './App.css';

function App() {
    const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
    const [error, setError] = useState<string | undefined>(undefined);

    useEffect(() => {
        extensionSetup()
    }, []);

    const extensionSetup = async () => {
        const extensions = await web3Enable('My Polkadot app');
        if (isWeb3Injected) {
          const accounts = await web3Accounts();
          setAccounts(accounts);          
        } else {
          setError("no extension found")
        }
    };

    return (
        <div className="App">
          {
            error && <div>Error: {error}</div>
          }
          {
            accounts.map(account => <div style={{ marginTop: 10}}>{account.address}</div>)
          }
        </div>
    );
}

export default App;
```

**Comment by jacogr** [comment]
It is injected, Chrom ensure the content scripts are added to all pages. However you need to ensure that the page has completed loading before attempting to access it. 

While the extension does wait for the document ready before firing (see https://github.com/polkadot-js/extension/blob/master/packages/extension-dapp/src/index.ts#L75, which mean that Chrome would have injected the content script, see https://github.com/polkadot-js/extension/blob/master/packages/extension/manifest.json#L18-L19) there may be a ms faster delay.

The way the apps UI ensures it is always available is to delay the init as long as possible, in that case even after the API is available, see https://github.com/polkadot-js/apps/blob/master/packages/react-api/src/Api.tsx#L215

In short, it is a browser timing issue, immediately at first paint the content script may not be fully ready.

**Comment by nilathj** [comment]
Thanks for your response, I'm using a useEffect hook which is run after the render of the component.  Inside my use effect, I'm using the steps in the cookbook: https://polkadot.js.org/docs/extension/cookbook/.  I have tried both the promise variant and the async await variant with the same result.  I agree its a timing issue, can you please point me in the right direction of a simple app which interacts with the polkadot js extension and works when the page is reloaded?


```
        const extensions = await web3Enable('My Polkadot app');
        if (isWeb3Injected) {
          const accounts = await web3Accounts();
```
Why is the async await and the fact that this is running inside a useEffect hook not sufficient?

**Comment by jacogr** [comment]
No idea, like I said it depend on page order injection (as controlled by the browser). I linked to the apps UI code as to how it controls it.

As always, PRs are welcome if there is a trick missing in the linked extension code.
